python/cudf/cudf/dataframe/datetime.py

```diff
@@ -154,7 +155,12 @@ def astype(self, dtype):
 
             return string.StringColumn(data=data)
 
-        return self.as_numerical.astype(dtype)
+        import cudf.bindings.typecast as typecast
+
+        col = typecast.apply_cast(
+            self.as_numerical, dtype=np.dtype(dtype).type
```

Comment:
Is there a reason we need to get this as a numerical column first? This will cause issues later with typecasting to / from different datetime types.

Reply:
Doesn't seem to be logical to do this now. Removing it.